src/fairchem/core/models/finetune_hydra.py

```diff
@@ -0,0 +1,187 @@
+from __future__ import annotations
+
+import copy
+import errno
+import logging
+import os
+from abc import ABC, abstractmethod
+from enum import Enum
+from typing import TYPE_CHECKING
+
+import torch
+from torch import nn
+
+from fairchem.core.common.registry import registry
+from fairchem.core.common.utils import load_state_dict, match_state_dict
+from fairchem.core.models.base import BackboneInterface, HeadInterface, HydraInterface
+
+if TYPE_CHECKING:
+    from torch_geometric.data import Batch
+
+FTHYDRA_NAME = "finetune_hydra"
+
+class FineTuneMode(Enum):
+    # in DATA_ONLY, we load the entire model and only finetune on new data
+    DATA_ONLY = 1
+    # in this mode, we only load the Backbone and feed the output of the backbone
+    # to new heads that are specified
+    RETAIN_BACKBONE_ONLY = 2
+
+
+def get_model_config_from_checkpoint(checkpoint_path: str) -> dict:
+    if not os.path.isfile(checkpoint_path):
+        raise FileNotFoundError(
+            errno.ENOENT, "Checkpoint file not found", checkpoint_path
+        )
+    checkpoint = torch.load(checkpoint_path)
+    return checkpoint["config"]["model"]
+
+
+def load_hydra_model(checkpoint_path: str) -> HydraInterface:
+    if not os.path.isfile(checkpoint_path):
+        raise FileNotFoundError(
+            errno.ENOENT, "Checkpoint file not found", checkpoint_path
+        )
+    logging.info(f"Loading checkpoint from: {checkpoint_path}")
+    checkpoint = torch.load(checkpoint_path)
+    config_copy = copy.deepcopy(checkpoint["config"]["model"])
```

Comment:
Do you need deepcopy here? The checkpoint['config'] is not used elsewhere outside the scope of this function?